presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorCodecProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.connector;
+
+import com.facebook.presto.spi.ConnectorCodec;
+import com.facebook.presto.spi.ConnectorDeleteTableHandle;
+import com.facebook.presto.spi.ConnectorInsertTableHandle;
+import com.facebook.presto.spi.ConnectorOutputTableHandle;
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.facebook.presto.spi.ConnectorTableLayoutHandle;
+
+import java.util.Optional;
+
+public interface ConnectorCodecProvider
```

Comment:
I don't think there's a use case for selectively enabling codecs for only some fields.  If that's the case, should this be changed to supply all fields, instead of making each one optional?  In which case, `Connector#getConnectorCodecProvider` would return `Optional<ConnectorCodecProvider>`, indicating just binary yes/no on whether or not a custom codec has been provided.

Reply:
Why don't they Thrift-ize all fields?